### ISSUE TITLE
all.js problem with packed plugins when override Shopware core functionality

### ISSUE THREAD

**Issue by arthur-ditegra** [FriendsOfShopware/shopware-cli#430]
### PHP Version

8.1

### Shopware Version

6.5.8.10

### Plugin Version

 0.4.54

### Actual behaviour

The Problem happens when installing the Plugin in an excising or newly created Shopware system. In this example, we override the off canvas cart:
`PluginManager.override('OffCanvasCart', OverOffcanvasCart, '[data-offcanvas-cart]');` An Error is thrown when the packet part reaches `o = l.Z.serialize(t);`
`l` should be the off canvas cart or an Object. I am not sure there.
This is not the first time it has happened, somehow the packet all.js has some wrong Id's set.

### Expected behaviour

The packed all.js should behave the same when packing a Plugins JavaScript into an all.js file.

### Steps to Reproduce?

* Just create a plugin and extend or override Shopware core code.

For this example, you can use this:

**main.js**
````javascript
import TestOffcanvasCart from './plugin/test-offcanvas-cart/test-offcanvas-cart.plugin';

const PluginManager = window.PluginManager;
PluginManager.override('OffCanvasCart', TestOffcanvasCart, '[data-offcanvas-cart]');
```` 

**test-offcanvas-cart.plugin.js.js**
````javascript
import OffcanvasCartPlugin from 'src/plugin/offcanvas-cart/offcanvas-cart.plugin';

export default class TestOffcanvasCartPlugin extends OffcanvasCartPlugin {

    _onChangeProductQuantity(event) {
        const select = event.target;
        const form = select.closest('form');
        const selector = this.options.cartItemSelector;
        this.$emitter.publish('onChangeProductQuantity');
        //
        const url = window.router['frontend.cart.offcanvas'];
        const _callback = () => {
            this.client.get(url, response => {
                this._updateOffCanvasContent(response);
                this._registerEvents();
            }, 'text/html');
        };
        this._fireRequest(form, selector, _callback);
    }
}
```` 
* Pack this plugin with the CLI tool
* Install the plugin and observer the log when you change the quantity of products inside the offcanvas-cart.

I hope this is enough information to reproduce.

**Comment by shyim** [comment]
what did you specified in the plugin as compatiblity for shopware version? By default it takes the oldest shopware version to build it with could break it

**Comment by arthur-ditegra** [comment]
Thanks for the reply @shyim.

This is in our compose.json:
````json "require": {
    "shopware/core": "~6.6.0",
    "shopware/storefront": "~6.6.0"
````
This means the CLI tool will use SW 6.6.0. What we found out in the docs.

For this reason, we fixed the used CLI version in our Pipeline to the current Shopware version:
````yml
 - name: Install shopware-cli
        uses: FriendsOfShopware/shopware-cli-action@v1
        with:
          version: ${{ env.SHOPWARE_CLI_VERSION }}
````
Some additional information:
We use the GitHub Actions version of the CLI tool.

PS.: I hope you will find a good replacement name for `shopware-cli` ;)

**Comment by shyim** [comment]
Can you maybe provide the zip file of the plugin?

**Comment by arthur-ditegra** [comment]
Yes, sure, I got the permission to do so. Can I DM you with the ZIP in Slack or via E-Mail?

**Comment by shyim** [comment]
Can you create a `.shopware-extension.yml` file in the root of the plugin and add something like this:

```
build:
  shopwareVersionConstraint: '6.5.8.10'
```

so Shopware CLI builds using that Shopware version instead picking the lowest which is here 6.5.0.0

**Comment by arthur-ditegra** [comment]
I will add the build constraint and observe what will happen.

**Comment by arthur-ditegra** [comment]
@shyim Hello again.

# SW 6.5
````
build:
  shopwareVersionConstraint: '6.5.8.10'`
````
The code above had pointed me to this:

I found out that the JS from the older JS version is not compatible!
````Jason
"require": {
    "shopware/core": "~6.5.8",
    "shopware/storefront": "~6.5.8"
  },
````
Changing the requirement to only support from version 6.5.8 helped me to fix the problem for SW 6.5

# SW 6.6
In SW 6.6, there was a different problem. As you see, we extend the `OffcanvasCartPlugin`
Since SW 6.6.6, a `focusHandler` was added. So building with this or newer version broke the plugin.

When installing this Plugin in an older SW Version like 6.6.4, the JavaScript throws an Error that the `focusHandler` is not defined.

But this is all Shopware Related and not this CLI-Tool. 

It would be nice to have a notification or a warning that the JS builded now is only Compatible to x version


Thanks again for the Help. Your commend leaded me to the right direction!

**Comment by shyim** [comment]
sounds good to have a warning, generally I would recommand to built for each major a new version 👍
